Improve build/example section of README.md

alexxcons opened the pull request:
- Simple build instruction, with reference to more advanced options
- Make clear that building opencmw-cpp itself is not required when the
intention is, to create a new service
- Reference to opencmw-cpp-example instead of Java example

RalphSteinhagen commented on README.md:
```diff
@@ -83,61 +83,36 @@ provides also an optional light-weight `constexpr` annotation template wrapper `
+### Building from source
 
-### Example
-The following provides some flavour of how a simple service can be implemented using OpenCMW with only a few lines of
```
This code example example is important and provides a flavour of the library and concepts involved.

While the syntax is different ins some parts, we want to keep the examples/documentation pretty much aligned between the C++ and Java version of OpenCMW.

Ideally, using language specific code tabs like:
![image](https://user-images.githubusercontent.com/46007894/160841853-b1108900-0937-4236-bbcc-f9ebc57100a1.png)

While GH markdown does not support this out-of-the box (N.B. to be further investigated), we should setup HTML doc  page where this is done, and at least keep/leave the corresponding C++ version here.

The code syntax flavour is more important than how to use `cmake` (which pretty standard anyway).

RalphSteinhagen replied:
To give a specific flavour of documentation style we want to emulate have a look at, for example: https://javalin.io/

RalphSteinhagen commented on README.md:
````diff
@@ -83,61 +83,36 @@ provides also an optional light-weight `constexpr` annotation template wrapper `
+Building on linux operating systems:
+
+```
+git clone https://github.com/alexxcons/opencmw-cpp.git
````
wrong github repo

RalphSteinhagen commented on README.md:
```diff
@@ -83,61 +83,36 @@ provides also an optional light-weight `constexpr` annotation template wrapper `
+git clone https://github.com/alexxcons/opencmw-cpp.git
+cd opencmw-cpp
+mkdir build
+ccmake -S . -B ./build
```
Aside being unnecessarily complex for uninitiated cmake users, we should not use UI-style and in-source build flavour cmake.

A simpler and more intuitive way which is also copy-paste safe would be:
```
cd opencmw-cpp
mkdir build; cd build
cmake .. -DCMAKE_BUILD_TYPE=Release # for debug builds use 'Debug'
make -j 4
```
...Execute tests via `ctest` in `.` and samples in `./concepts/<core|cmrc|disruptor|majordomo|serialiser>/`...

Ideally this would be also wrapped in a by-language code tab to keep the doc in sync.

alexxcons replied:
Alright .. I just tried to use what was suggested in `docs/BuildInstructions.md`.

RalphSteinhagen commented on README.md:
````diff
@@ -83,61 +83,36 @@ provides also an optional light-weight `constexpr` annotation template wrapper `
+* If there are no errors, press 'g' and wait until 'generate' is done and start the build:
+
+```
+cmake --build ./build
````
... can cause confusion and needless complex cmd-line syntax

RalphSteinhagen commented on README.md:
````diff
@@ -83,61 +83,36 @@ provides also an optional light-weight `constexpr` annotation template wrapper `
+cmake --build ./build
+```
+
+For detailed, advanced options or installation on non-linux operating systems, please check the [detailed build instructions](docs/BuildInstructions.md).
````
is probably the better place to add these cmake snippets

alexxcons replied:
Ok, will move all the snippets to `docs/BuildInstructions.md` and only leave a link here.

alexxcons commented:
Fixed now, please re-check.

- Moved all build instruction snippets into BuildInstructions.md
- I did not find a way to use language specific code tabs in github, so that will be for later.
- I just kept the current java example as it is for now. If I find time, I will try to add the same example in C++

RalphSteinhagen reviewed (approved):
Thanks for the changes! LGTM! :+1:
